Mlem/App/Environment Values/CommunityContext.swift

```diff
@@ -0,0 +1,20 @@
+//
+//  CommunityContext.swift
+//  Mlem
+//
+//  Created by Eric Andrews on 2024-05-22.
+//
+
+import MlemMiddleware
+import SwiftUI
+
+struct CommunityContext: EnvironmentKey {
+    static let defaultValue: (any Community3Providing)? = nil
```

Comment:
Sure--I had originally done it with Community3 to enforce that the model is upgraded because community context is largely useless without the moderators list, so this way we'll just have to ensure that it's handled view-side ourselves.